### ISSUE TITLE
Catch2 warning suppression macros changed 10/27/2019

### ISSUE THREAD

**Issue by codusnocturnus** [eranpeer/FakeIt#197]
Catch2 changed their approach to warning suppression macros such that the "UNSUPPRESS" macros are no longer defined, which breaks compilation for single_header/catch/fakeit.hpp at line 1225.

Read the commit message here for details:
https://github.com/catchorg/Catch2/commit/849f2848bdf625a7b6248e4e62ba0e39ee609634#diff-a010dcdfe8ba1736a2dc3acde52c3e89

Compilation succeeds if I change CatchAdapter::Fail() to:
```
void fail(
        std::string vetificationType,
        Catch::SourceLineInfo sourceLineInfo,
        std::string failingExpression,
        std::string fomattedMessage,
        Catch::ResultWas::OfType resultWas = Catch::ResultWas::OfType::ExpressionFailed ){
    Catch::AssertionHandler catchAssertionHandler( vetificationType, sourceLineInfo, failingExpression, Catch::ResultDisposition::Normal );
    INTERNAL_CATCH_TRY { \
        CATCH_INTERNAL_START_WARNINGS_SUPPRESSION \         // added line
        CATCH_INTERNAL_SUPPRESS_PARENTHESES_WARNINGS \
        catchAssertionHandler.handleMessage(resultWas, fomattedMessage); \
        CATCH_INTERNAL_STOP_WARNINGS_SUPPRESSION \         // changed line
    } INTERNAL_CATCH_CATCH(catchAssertionHandler) { \
        INTERNAL_CATCH_REACT(catchAssertionHandler) \
    }
}
```

**Comment by nniles** [comment]
Covered by PR #195

**Comment by shrgabor** [comment]
Dear @eranpeer 
Can you please have a look on one of these PR's? #201 and #195 . They are trying to solve the same problem. And they are opened for quite a long time.
I think it would worth a new release.
Anyhow, thank you for your fantastic framework. I really love it.

**Comment by pgorgon-hem** [comment]
I fixed PR #195, please merge it, there are some people who need it.

**Comment by Robbie-Palmer** [comment]
I'm using Catch v2.12.1 and received the error `error: ‘CATCH_INTERNAL_UNSUPPRESS_PARENTHESES_WARNINGS’ was not declared in this scope` on compilation
The referenced changes fixed it for me

**Comment by sd185233** [comment]
Are there any plans to release this fix?

**Comment by sdaly2107** [comment]
Are there any plans to release this fix?

**Comment by pharapeti** [comment]
> I'm using Catch v2.12.1 and received the error `error: ‘CATCH_INTERNAL_UNSUPPRESS_PARENTHESES_WARNINGS’ was not declared in this scope` on compilation
> The referenced changes fixed it for me

Same here

**Comment by zfields** [comment]
I'm on Catch `v2.13.4` and I can confirm this is still an issue. Just chiming in to increase visibility. Thanks, Zak. I looked over #195 and it looks like a quick and easy PR to fix the issue. It's a 6 line change that will help a lot of people use FakeIt. Cheers.

**Comment by FranckRJ** [comment]
Fixed in 2.0.6.
